### DISCUSSION TITLE
workflow for multi-package development

### DISCUSSION THREAD

**Discussion by GooseYArd** [pdm-project/pdm#611] [Q&A] [answered]
I'm almost certain that I found a post someplace regarding the scenario I'm about to describe, although I can't find it in my history, and I may be remembering a Poetry-related post.

I manage a package which contained a handful of distinct modules, and recently it became necessary to refactor those modules into discreet packages which can be released independently. Predictably, on occasion a change in one module will require a corresponding change in another, and so it's convenient to use filesystem path dependencies in the pyproject.toml during development.

Currently I'm doing this in the laziest possible way- I just comment out the name-based dependencies and uncomment a block of filesystem path dependencies. This works but looks a little messy and it's easy to forget the put the right dependencies back before locking.

I suspect that there's a nicer way to manage this, and thought I'd here to see if someone has a clever workflow for scenarios like mine. 

Thanks!!

**Comment by pawamoy** [comment, marked as answer]
I think I remember seeing a commit bringing precedence to dependencies, so that developers can specify both the "named" one *and* the editable, local one (`-e relative_path`). @frostming could probably confirm or deny this.

**Reply by frostming** [reply]
True, editable dependencies will take precedence over named ones when doing `pdm install` and `pdm sync`, but they will be ignored when building the wheel -- named ones will be written to the metadata(and uploaded to PyPI afterward).

**Reply by GooseYArd** [reply]
OK I officially love pdm now.
